feat: Display view switcher for recent files

doubleface opened the pull request:
And also allow to hide sorter when canSort is not true.

<img width="337" height="584" alt="image" src="https://github.com/user-attachments/assets/d405fffd-4f5a-4999-8191-8a005c8e3879" />



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * Mobile file list header layout corrected to preserve right-side alignment while properly showing sort controls.

* **New Features**
  * Centralized file-sorting applied across Folder and Recent views for consistent, stable sorting and selection behavior.

* **Tests**
  * Added comprehensive tests covering sorting logic, default and external sort orders, secondary sorting, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
feat: Allow to hide sorter on mobile

And always display the view switcher

rezk2ll commented on src/modules/filelist/FileListHeaderMobile.jsx:
```diff
@@ -36,13 +37,14 @@ const FileListHeaderMobile = ({
   )
 
-  if (!canSort) return null
+  if (!canSort && !canSwitchViewType) return null
```
i don't think we need to have this `canSwitchViewType` here, no matter what mobile or desktop we should be able to switch the view

rezk2ll commented on src/modules/filelist/FileListHeaderMobile.jsx:
```diff
@@ -63,6 +65,7 @@ const FileListHeaderMobile = ({
           />
         )}
         <TableHeader
+          style={{ visibility: canSwitchViewType ? 'visible' : 'hidden' }}
```
it should be always displayed

doubleface replied:
Yes, but at first the only criteria to display the header was if canSort is thruthy. Here, for the recents view, I want to display the header but only with the "change view type" button. 
Yes, at the moment, "change view type" button will always be display, but canSwitchViewType gives me a new criteria to display the mobile header or not and allowing to hide it make the api more coherent I think.

doubleface replied:
Or should we change the logic and add a "hideHeader" prop to FolderViewBody ?

doubleface replied:
Let's always show the header with the view switch and hide the sorter when canSort is not true

rezk2ll commented:
i don't understand why this PR is needed

Commit:
refactor:  Extract file sorting logic from FolderViewBody

Introduced `useFileSorting` hook to encapsulate file sorting
functionality, improving code organization and reusability. Updated
`FolderViewBody` to utilize the new hook for sorting files, enhancing
maintainability. Added tests for the new hook to ensure correct
behavior.

doubleface commented:
> i don't understand why this PR is needed

This PR is needed to display the view switcher on the recents file view.

I am sorry because at first I also added the sort feature because it was the only way to display the toolbar.
After some thinking, I prefered to make it possible to display only the view switcher in the toolbar, without the sorting.

doubleface commented on src/modules/views/Folder/hooks/useFileSorting.spec.js:
```diff
@@ -0,0 +1,136 @@
+
+    expect(result.current.sortOrder).toEqual({
+      order: 'desc',
+      attribute: 'updated_at'
+    })
+  })
+
+  it('should apply secondary sort when no sort order specified', () => {
+    const { result } = renderHook(() =>
+      useFileSorting(
+        'folder-1',
+        [
+          {
+            data: [
+              { _id: '1', name: 'file.txt', type: 'file' },
+              { _id: '2', name: 'folder', type: 'directory' }
+            ]
+          }
+        ],
+        {}
+      )
+    )
+
+    // Secondary sort should put directories before files
+    expect(result.current.sortedFiles[0].type).toBe('directory')
+    expect(result.current.sortedFiles[1].type).toBe('file')
+  })
+
+  it('should apply both primary and secondary sort', () => {
+    const { result } = renderHook(() =>
+      useFileSorting('folder-1', mockQueryResults, mockOrderProps)
+    )
+
+    // Should be sorted by name in descending order, then by type
+    expect(result.current.sortedFiles).toHaveLength(3)
+  })
```
Added some tests on sorting

doubleface commented on src/modules/views/Folder/hooks/useFileSorting.spec.js:
```diff
@@ -0,0 +1,161 @@
+    expect(result.current.sortedFiles[1].type).toBe('file')
+  })
+
+  it('should apply both primary and secondary sort', () => {
+    const { result } = renderHook(() =>
+      useFileSorting('folder-1', mockQueryResults, mockOrderProps)
+    )
+
+    // Should be sorted by name in descending order, then by type
+    expect(result.current.sortedFiles).toHaveLength(3)
+
+    // Verify the actual sort ordering
+    const sortedFiles = result.current.sortedFiles
+
+    // Expected order: name descending (folder-c, file-b.txt, file-a.txt)
+    // Secondary sort: directories before files (folder-c should come before files)
+    expect(sortedFiles[0].name).toBe('folder-c')
+    expect(sortedFiles[0].type).toBe('directory')
+
+    expect(sortedFiles[1].name).toBe('file-b.txt')
+    expect(sortedFiles[1].type).toBe('file')
+
+    expect(sortedFiles[2].name).toBe('file-a.txt')
+    expect(sortedFiles[2].type).toBe('file')
+
+    // Alternative verification: map filenames and types
+    const fileInfo = sortedFiles.map(file => ({
+      name: file.name,
+      type: file.type
+    }))
+    expect(fileInfo).toEqual([
+      { name: 'folder-c', type: 'directory' },
+      { name: 'file-b.txt', type: 'file' },
+      { name: 'file-a.txt', type: 'file' }
+    ])
+  })
```
done

doubleface commented on src/modules/views/Folder/hooks/useFileSorting.spec.js:
```diff
@@ -0,0 +1,161 @@
+    // Secondary sort should put directories before files
+    expect(result.current.sortedFiles[0].type).toBe('directory')
+    expect(result.current.sortedFiles[1].type).toBe('file')
+  })
```
done
